docs/decisions/0005-architecture-and-data-modeling.rst

```diff
@@ -0,0 +1,149 @@
+- The Open edX Layer will provide a stable Enforcement API that abstracts Casbin internals, allowing services to request authorization decisions without needing to understand Casbin specifics.
+- Implement a Policy Management API within the Open edX Layer to allow administrators to manage and update authorization policies centrally.
+- The Open edX Layer will be implemented as a shared library, a Django app or a Tutor plugin or all three, depending on the what is most suitable. All of these components will be versioned and maintained as part of the Open edX ecosystem as a whole.
+- All modifications to the Authorization Engine configuration (model, adapters, etc.) should be done through the Open edX Layer, so no forks of Casbin are needed.
+
+Interact with the Policy Store via the Open edX Layer
+------------------------------------------------------
+- The policy store, which can be a database or configuration files (e.g., ``authz.policy``) will be accessed and managed through the Open edX Layer. No direct access to the policy store should be made by services.
+- The Open edX Layer will handle loading policies from the policy store into the Authorization Engine and ensure that policies are kept up to date.
+- The Open edX Layer will also manage the separation between static policies (shipped with services) and dynamic policies (managed via the policy data store) to ensure clarity and maintainability.
+
+Clients Interact Only via the Open edX Layer
+--------------------------------------------
+- Services and other clients (e.g., MFEs) will interact with the authorization system exclusively through the Open edX Layer's Enforcement API.
+- Services will not implement their own authorization logic or interact directly with Casbin internals.
+- Services can request authorization decisions by calling the Enforcement API with the necessary context (e.g., user, action, resource) with optional context and receiving a decision (allow/deny) in response.
+
+#. Data & Storage Model
+========================
+
+Use a MySQL as the Main Backend for the Policy Store
+----------------------------------------------------
+- Use MySQL as the main backend for persistent storage for policies (policy store), leveraging Casbin's Django ORM adapter for integration and our own modifications to the adapter as needed in our own ``engine/``` package.
+- Use the same schema used by Casbin's Django ORM adapter, which includes a table for storing policies with:
+  - ``id``: A unique identifier for each policy. This depends on the database backend. In our case, MySQL uses an auto-incrementing integer.
+  - ``ptype``: The policy type (e.g., ``p`` for policy, ``g, g2`` for grouping).
+  - ``v0, v1, v2, v3, v4, v5``: The policy fields that define the subject, action, object, and context. The exact meaning of these fields depends on the policy type and model configuration.
+- Optionally, include additional metadata fields in the policy table to support auditing and versioning.
```

Comment:
I think these are two different things? We should have as part of this decision what we want to store for auditing and versioning, and it should be non-optional. Whereas metadata fields may be optional?